src/views/SignUp/SignUpComponents/SignUpComponent.vue

```diff
@@ -1,71 +1,158 @@
 <script setup lang="ts">
-import { ref } from 'vue';
+import { ref, computed } from 'vue';
 import { registerUserThroughFirebase } from '@/auth/auth';
-import { getAuth, signOut } from 'firebase/auth';
 import router from '@/router';
-import authService from "@/services/auth.service";
-import {firebaseApp} from "@/firebase";
+import authService from '@/services/auth.service';
 
+// Form inputs
 const displayName = ref('');
 const email = ref('');
 const password = ref('');
 const confirmPassword = ref('');
 
+// Computed validation logic
+const usernameError = computed(() => {
+  if (!displayName.value) return 'Username is required';
+  if (!/^[a-zA-Z0-9]{4,}$/.test(displayName.value)) {
+    return 'Username must be at least 4 characters and only contain letters and numbers';
+  }
+  return null;
+});
 
-const handleSignUp = async () => {
-  const auth = getAuth();
-  if (
-    displayName.value.trim() === '' ||
-    email.value.trim() === '' ||
-    password.value.trim() === '' ||
-    confirmPassword.value.trim() === ''
-  ) {
-    alert('Please fill in all fields.');
-    return;
+const passwordError = computed(() => {
+  if (!password.value) return 'Password is required';
+  if (!/^[a-zA-Z!@#$%^&*(),.?":{}|<>]{8,}$/.test(password.value)) {
+    return 'Password must be at least 8 characters and contain letters or special characters';
   }
+  return null;
+});
+
+const confirmPasswordError = computed(() => {
   if (password.value !== confirmPassword.value) {
-    alert('Passwords do not match.');
+    return 'Passwords do not match';
+  }
+  return null;
+});
+
+const isEmailValid = computed(() => /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email.value));
+
+const isFormValid = computed(
+  () =>
+    !usernameError.value &&
+    isEmailValid.value &&
+    !passwordError.value &&
+    !confirmPasswordError.value,
+);
+
+// Form submission
+const handleSignUp = async () => {
+  if (!isFormValid.value) {
+    alert('Please fix the errors before submitting.');
     return;
   }
-  // Successful sign up should redirect user to login page
-  const { uid } = await registerUserThroughFirebase(email.value, password.value, displayName.value);
-  // We need to add this to the db
-  const response = await authService.createUser(uid);
-  if (response.status === 201) {
-    await router.push({ name: 'login' });
-  } else {
-    // Error handling here
-    // Maybe we can delete the user
+
+  try {
+    const { uid } = await registerUserThroughFirebase(
+      email.value,
+      password.value,
+      displayName.value,
+    );
+
+    const response = await authService.createUser(uid);
+    if (response.status === 201) {
+      await router.push({ name: 'login' });
+    } else {
+      alert('Failed to create user in the database.');
+    }
+  } catch (error) {
+    console.error('Sign-up error:', error);
+    alert('An error occurred during sign-up. Please try again.');
```

Comment:
Maybe one more very little improvement is to figure out a better way to give users feedback instead of an alert. I remember the feedback about alerts yesterday was a bit negative.